enterprise/server/workflow/service/service.go

```diff
@@ -492,7 +485,59 @@ func (ws *workflowService) ExecuteWorkflow(ctx context.Context, req *wfpb.Execut
 		return nil, err
 	}
 
-	// Fetch the workflow config for the action we're about to execute
+	actions, err := ws.getActions(ctx, wf, wd, req.GetActionName())
+	if err != nil {
+		return nil, err
+	}
+
+	eg, egCtx := errgroup.WithContext(ctx)
+	actionToInvocationID := make(map[string]string, len(actions))
+	for _, action := range actions {
+		action := action
+		eg.Go(func() error {
```

Comment:
it might make sense to use a regular `WaitGroup` for this so that we can allow some actions to be started and some to fail (e.g. if they try to execute with `os: darwin` and no mac executors are registered). Then we can report a google.grpc.Status for each started action, with the error code

Reply:
Done. Added a status_code to the Action struct